narwhal/consensus/src/bullshark.rs

```diff
@@ -35,42 +53,30 @@ impl ConsensusProtocol for Bullshark {
         let round = certificate.round();
         let mut consensus_index = consensus_index;
 
-        // We must have stored already the parents of this certiciate!
-        if round > 0 {
-            let parents = certificate.header.parents.clone();
-            if let Some(round_table) = state.dag.get(&(round - 1)) {
-                let store_parents: BTreeSet<&CertificateDigest> =
-                    round_table.iter().map(|(_, (digest, _))| digest).collect();
-
-                for parent_digest in parents {
-                    if !store_parents.contains(&parent_digest) {
-                        if round - 1 + self.gc_depth > state.last_committed_round {
-                            error!(
-                                "The store does not contain the parent of {:?}: Missing item digest={:?}",
-                                certificate, parent_digest
-                            );
-                        } else {
-                            debug!(
-                                "The store does not contain the parent of {:?}: Missing item digest={:?} (but below GC round)",
-                                certificate, parent_digest
-                            );
-                        }
-                    }
-                }
-            } else {
-                error!(
-                    "Round not present in Dag store: {:?} when looking for parents of {:?}",
-                    round - 1,
-                    certificate
-                );
-            }
-        }
+        // We must have stored already the parents of this certificate!
+        self.check_parents_exist(&certificate, state);
```

Comment:
What does "check" mean in the name here? It is not returning a value, so the check seems to have no actual function in the code? maybe "log_error_if_missing_parents" would be a better name that makes it very clear no actual action is taken?

(I'll admit to some Google naming bias here, because at Google the word "check" means the same as "assert" - the program will die if the precondition is not met. But this doesn't have any asserts or panics.)

Reply:
Curious to know why we allow certificate to continue to process and only print the error? Shouldn't we just stop here if cert parents must be stored?

Reply:
@aschran good point. I wasn't 100% sure with the "check" either. Usually my approach is that `ensure_....`  will actually panic/return error, so next thing came to my mind was "check". `log_error_if_missing_parents` is much more appropriate

Reply:
@arun-koshy yes you are right - and that was the first iteration when that code was introduced. But it was probably considered quite invasive to panic - so wanted first to confirm that this is actually happening so that's why ended up using logging